diff --git a/pyproject.toml b/pyproject.toml
index 6ef699d..32e4bc3 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,7 +16,7 @@ authors = [
 ]
 description = "A python CAD programming library"
 readme = "README.md"
-requires-python = ">= 3.10, < 3.13"
+requires-python = ">= 3.10, < 3.14"
 keywords = [
     "3d models",
     "3d printing",
@@ -35,7 +35,7 @@ classifiers = [
 ]
 
 dependencies = [
-    "cadquery-ocp >= 7.7.0",
+    "cadquery-ocp >= 7.8.1",
     "typing_extensions >= 4.6.0, <5",
     "numpy >= 2, <3",
     "svgpathtools >= 1.5.1, <2",
diff --git a/src/build123d/topology.py b/src/build123d/topology.py
index 5986e08..ec9c586 100644
--- a/src/build123d/topology.py
+++ b/src/build123d/topology.py
@@ -969,7 +965,7 @@ class Shape(NodeMixin, Generic[TOPODS]):
         """
         if self.wrapped is None:
             return 0
-        return self.wrapped.HashCode(HASH_CODE_MAX)
+        return hash(self.wrapped)
 
     def is_null(self) -> bool:
         """Returns true if this shape is null. In other words, it references no
@@ -9612,7 +9612,7 @@ def _topods_entities(shape: TopoDS_Shape, topo_type: Shapes) -> list[TopoDS_Shap
 
     while explorer.More():
         item = explorer.Current()
-        out[item.HashCode(HASH_CODE_MAX)] = (
+        out[hash(item)] = (
             item  # needed to avoid pseudo-duplicate entities
         )
         explorer.Next()
