diff --git a/tests/test_assembly.py b/tests/test_assembly.py
index bae54ab..1093ffb 100644
--- a/tests/test_assembly.py
+++ b/tests/test_assembly.py
@@ -702,20 +702,17 @@ def test_save(extension, args, nested_assy, nested_assy_sphere):
         ("stl", ("STL",), {}),
     ],
 )
-def test_export(extension, args, kwargs, tmpdir, nested_assy):
+def test_export(extension, args, kwargs, nested_assy):
 
     filename = "nested." + extension
-
-    with tmpdir:
-        nested_assy.export(filename, *args, **kwargs)
-        assert os.path.exists(filename)
+    nested_assy.export(filename, *args, **kwargs)
+    assert os.path.exists(filename)
 
 
-def test_export_vtkjs(tmpdir, nested_assy):
+def test_export_vtkjs(nested_assy):
 
-    with tmpdir:
-        nested_assy.export("nested.vtkjs")
-        assert os.path.exists("nested.vtkjs.zip")
+    nested_assy.export("nested.vtkjs")
+    assert os.path.exists("nested.vtkjs.zip")
 
 
 def test_export_errors(nested_assy):
