--- CMakeLists.txt	2025-09-18 18:49:41
+++ CMakeLists.txt.patched	2025-09-18 18:50:17
@@ -4,15 +4,6 @@


 find_package( Python REQUIRED COMPONENTS Development Interpreter)
-find_package( VTK REQUIRED
-  COMPONENTS
-    WrappingPythonCore
-    RenderingCore
-    RenderingOpenGL2
-    CommonDataModel
-    CommonExecutionModel
-    freetype
-)
 find_package( pybind11 REQUIRED )
 find_package( OpenCASCADE REQUIRED )

@@ -24,10 +15,6 @@
   ${OpenCASCADE_LIBRARIES}
   Python::Module
   pybind11::pybind11
-  VTK::WrappingPythonCore
-  VTK::RenderingCore
-  VTK::CommonDataModel
-  VTK::CommonExecutionModel
   )
 set_target_properties( OCP
                        PROPERTIES