diff --git a/bindgen/__init__.py b/bindgen/__init__.py
index e08bed9..a03c7a0 100644
--- a/bindgen/__init__.py
+++ b/bindgen/__init__.py
@@ -45,8 +45,9 @@ def read_symbols(p):
     This information is used later for flagging undefined symbols
     '''
 
-    sym = pd.read_csv(p,header=None,names=['name'],delim_whitespace=True,
-                      error_bad_lines=False).dropna()
+    sym = pd.read_csv(
+        p, header=None, names=["name"], sep='\s+', on_bad_lines="warn"
+    ).dropna()
     return sym
 
 def remove_undefined_mangled(m,sym):
